setup.py

```diff
@@ -127,15 +129,16 @@ def is_platform_windows():
     "install_requires": [
         "numpy>=1.18",
         "scipy",
-        "astropy>=3.2.3",
+        "astropy>=4.2",
```

Comment:
We should talk about whether we should roll the other changes that need to happen when we require astropy 4.2 into this PR or not. See #955

Reply:
I think those changes should be rolled in -- I'll plan on incorporating before I move the PR out of draft.

Reply:
Done.